src/eva/vision/data/datasets/classification/mhist.py

```diff
@@ -74,16 +69,16 @@ def validate(self) -> None:
         )
 
     @override
-    def load_image(self, index: int) -> np.ndarray:
+    def load_image(self, index: int) -> tv_tensors.Image:
         image_filename, _ = self._samples[index]
         image_path = os.path.join(self._dataset_path, image_filename)
-        return io.read_image(image_path)
+        return io.read_image_as_tensor(image_path)
 
     @override
-    def load_target(self, index: int) -> np.ndarray:
+    def load_target(self, index: int) -> torch.Tensor:
         _, label = self._samples[index]
         target = self.class_to_idx[label]
-        return np.asarray(target, dtype=np.int64)
+        return torch.tensor(target, dtype=torch.float32)
```

Comment:
In CRC we use `torch.long` and here `torch.float32`. What is the reason for that?

Reply:
its for binary classification (mhist and pcam), the criterion need a float from 0 - 1; currently we transform the type to float during transforms but I figured since we will do it either-way maybe we can do straight from the dataset